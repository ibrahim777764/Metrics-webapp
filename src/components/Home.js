import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch, useSelector, shallowEqual } from 'react-redux';
import { FaRegArrowAltCircleRight } from 'react-icons/fa';
import { NavLink } from 'react-router-dom';
import CountryPreview from './CountryPreview';
import ErrorBoundary from './ErrorBoundary';
import { endSearch } from '../redux/search/search';

const Home = ({ theme, countries }) => {
  const dispatch = useDispatch();
  const search = useSelector((state) => state.search, shallowEqual);

  const { hsl } = theme;
  const { searchOpen, searchResult } = search;

  const handleClose = () => {
    dispatch(endSearch());
  };

  return (
    <main className="h-full w-full relative top-16 z-0 grid grid-cols-2 sm:grid-cols-4 lg:grid-cols-6 xl:grid-cols-8">
      {
        countries.map((country) => (
          <CountryPreview
            key={country.cca2.toLowerCase()}
            hsl={hsl}
            country={country}
          />
        ))
      }
      <ErrorBoundary>
        {
          searchOpen && (searchResult.length !== 0) && (
            <section className="absolute rounded overflow-hidden shadow-2xl w-10/12 md:w-5/12 z-20 top-3 right-3 bg-white text-blue-500 flex flex-col">
              {
                searchResult.map((country) => (
                  <NavLink
                    to={`/${country.name.common.toLowerCase()}/details`}
                    className="flex justify-between even:bg-slate-100 p-2 items-center"
                    key={country.cca2}
                    onClick={handleClose}
                  >
                    <span>
                      {country.name.common}
                      {', '}
                      {country.capital[0]}
                    </span>
                    <FaRegArrowAltCircleRight className="text-base ml-1" />
                  </NavLink>
                ))
              }
            </section>
          )
        }
      </ErrorBoundary>
    </main>
  );
};

Home.propTypes = {
  theme: PropTypes.instanceOf(Object).isRequired,
  countries: PropTypes.instanceOf(Array).isRequired,
};

export default Home;
