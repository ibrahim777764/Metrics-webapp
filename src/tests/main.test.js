import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/store/configureStore';
import Home from '../components/main';

describe('Main Component test', () => {
  it('Renders main Page', () => {
    const app = renderer
      .create(
        <Provider store={store}>
          <Home />
        </Provider>,
      )
      .toJSON();
    expect(app).toMatchSnapshot();
  });
});
