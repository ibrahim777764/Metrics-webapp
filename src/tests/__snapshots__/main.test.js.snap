// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Component test Renders main Page 1`] = `
<div>
  <form>
    <input
      format="YYYY-MM-DD"
      id="date"
      type="date"
    />
    <button
      id="sub"
      onClick={[Function]}
      type="button"
    >
      CLICK!
    </button>
    <br />
    <input
      className="search"
      onChange={[Function]}
      placeholder="SEARCH BY COUNTRY NAME"
      type="text"
      value=""
    />
  </form>
  <div
    className="title"
  >
     
  </div>
</div>
`;
